compio-buf/src/io_buf.rs

```diff
@@ -88,265 +64,126 @@ pub unsafe trait IoBuf: 'static {
     {
         use std::ops::Bound;
 
+        let len = self.buf_len();
+
         let begin = match range.start_bound() {
             Bound::Included(&n) => n,
             Bound::Excluded(&n) => n + 1,
             Bound::Unbounded => 0,
         };
 
-        assert!(begin <= self.buf_capacity());
+        assert!(begin <= len);
```

Comment:
The `slice` method cannot slice the uninit space, right? What if I want to do so?

Reply:
Now we are not panicking if start or end overflow. It's OK but I think it should be documented.